apps/token/src/routes/governance/propose/update-asset/propose-update-asset.tsx

```diff
@@ -0,0 +1,190 @@
+
+  const onSubmit = async (fields: UpdateAssetProposalFormFields) => {
+    await submit({
+      rationale: {
+        title: fields.proposalTitle,
+        description: fields.proposalDescription,
+      },
+      terms: {
+        updateAsset: {
+          ...JSON.parse(fields.proposalTerms),
+        },
+        closingTimestamp: getClosingTimestamp(fields.proposalVoteDeadline),
+        enactmentTimestamp: getEnactmentTimestamp(
+          fields.proposalVoteDeadline,
+          fields.proposalEnactmentDeadline
+        ),
+      },
+    });
+  };
+
+  return (
+    <AsyncRenderer
+      loading={networkParamsLoading}
+      error={networkParamsError}
+      data={params}
+    >
+      <Heading title={t('UpdateAssetProposal')} />
+      <VegaWalletContainer>
+        {() => (
+          <>
+            <ProposalFormMinRequirements
+              minProposerBalance={
+                params.governance_proposal_updateAsset_minProposerBalance
+              }
+              spamProtectionMin={params.spam_protection_proposal_min_tokens}
+            />
+
+            {VEGA_DOCS_URL && (
+              <p className="text-sm" data-testid="proposal-docs-link">
+                <span className="mr-1">{t('ProposalTermsText')}</span>
+                <Link
+                  href={`${VEGA_DOCS_URL}/tutorials/proposals${docsLink}`}
+                  target="_blank"
+                >{`${VEGA_DOCS_URL}/tutorials/proposals${docsLink}`}</Link>
+              </p>
+            )}
+
+            {VEGA_EXPLORER_URL && (
+              <p className="text-sm">
+                {t('MoreAssetsInfo')}{' '}
+                <Link
+                  href={`${VEGA_EXPLORER_URL}/assets`}
+                  target="_blank"
+                >{`${VEGA_EXPLORER_URL}/assets`}</Link>
+              </p>
+            )}
+
+            <div data-testid="update-asset-proposal-form">
+              <form onSubmit={handleSubmit(onSubmit)}>
+                <ProposalFormSubheader>
+                  {t('ProposalRationale')}
+                </ProposalFormSubheader>
+
+                <ProposalFormTitle
+                  registerField={register('proposalTitle', {
+                    required: t('Required'),
+                  })}
+                  errorMessage={errors?.proposalTitle?.message}
+                />
+
+                <ProposalFormDescription
+                  registerField={register('proposalDescription', {
+                    required: t('Required'),
+                  })}
+                  errorMessage={errors?.proposalDescription?.message}
+                />
+
+                <ProposalFormSubheader>
+                  {t('UpdateAsset')}
+                </ProposalFormSubheader>
+
+                <ProposalFormTerms
+                  registerField={register('proposalTerms', {
+                    required: t('Required'),
+                    validate: {
+                      validateJson: (value) => {
+                        try {
```

Comment:
don't we have an isJson util? Or is that only for the block explorer JSON bits?

Reply:
We also have libs/react-helpers/src/lib/validate/index.ts could you put whatever function you end up using there? Also update other inputs that use this validation?

Reply:
@dexturr We used to have a poorly named 'isJson' util, but actually its use was for checking whether data was suitable for being displayed with the syntax highlighter - not only did it check it was valid json, but also checked whether the object had keys. It got renamed to 'suitableForSyntaxHighlighter'.

@mattrussell36 good shout, I'll move 'validateJson' there. Thanks